tests/components/teslemetry/test_config_flow.py

```diff
@@ -86,3 +88,102 @@ async def test_form_errors(hass: HomeAssistant, side_effect, error, mock_test) -
         CONFIG,
     )
     assert result3["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def test_reauth(hass: HomeAssistant, mock_test) -> None:
+    """Test reauth flow."""
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONFIG,
+    )
+    mock_entry.add_to_hass(hass)
+
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": mock_entry.entry_id,
+        },
+        data=CONFIG,
+    )
+
+    assert result1["type"] is FlowResultType.FORM
+    assert result1["step_id"] == "reauth_confirm"
+    assert not result1["errors"]
+
+    with patch(
+        "homeassistant.components.teslemetry.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            CONFIG,
```

Comment:
It's only a test but change something here and not use the same config which is already in place in the config entry so we can see it changes

Reply:
Done